main/src/ca/uwaterloo/flix/language/phase/CodegenHelper.scala

```diff
@@ -94,25 +94,73 @@ object CodegenHelper {
     */
   case class FlixClassName(ref: List[String]) extends QualName
 
+
+  /**
+    * This trait will get extended by all classes and interfaces that represent an enum case.
+    * All these classes include a field representing `tag`
+    */
+  sealed trait EnumCaseName extends QualName {
+    def tag : String
+  }
+
   /**
-    * Qualified name of an enum class
+    * Qualified name of an enum case interface
     * @param sym symbol of the enum
     * @param tag tag of the enum
     * @param tpe tpe of the field of the enum
     */
-  case class EnumClassName(sym: EnumSym, tag: String, tpe: WrappedType) extends QualName {
+  case class EnumCaseInterfaceName(sym: EnumSym, tag: String, tpe: WrappedType) extends EnumCaseName {
     val ref: List[String] = tpe match {
-      case WrappedPrimitive(t) => CodegenHelper.fixedEnumPrefix ::: sym.namespace ::: List(sym.name, typeSpecifier(t), tag)
-      case _ => CodegenHelper.fixedEnumPrefix ::: sym.namespace ::: List(sym.name, "object", tag)
+      case WrappedPrimitive(t) => CodegenHelper.fixedEnumPrefix ::: sym.namespace ::: List(sym.name, "interf", typeSpecifier(t), tag)
+      case _ => CodegenHelper.fixedEnumPrefix ::: sym.namespace ::: List(sym.name, "interf", tag)
     }
   }
 
   /**
+    * Qualified name of an enum case class
+    * @param sym symbol of the enum
+    * @param tag tag of the enum
+    * @param tpe tpe of the field of the enum
+    */
+  case class SECClassName(sym: EnumSym, tag: String, tpe: WrappedType) extends EnumCaseName {
+    val ref: List[String] = tpe match {
+      case WrappedPrimitive(t) => CodegenHelper.fixedEnumPrefix ::: sym.namespace ::: List(sym.name, "clazz", typeSpecifier(t), tag)
+      case _ => CodegenHelper.fixedEnumPrefix ::: sym.namespace ::: List(sym.name, "clazz", "obj", tag)
+    }
+  }
+
+  /**
+    * Qualified name of an enum case class
+    * @param sym symbol of the enum
+    * @param tag tag of the enum
+    * @param fields tpe of the field of the enum
+    */
+  case class ETFClassName(sym: EnumSym, tag: String, fields: List[WrappedType]) extends EnumCaseName {
+    private val desc: List[String] = fields.map{
+      case WrappedPrimitive(t) => typeSpecifier(t)
+      case _ => "obj"
+    }
+
```

Comment:
I am confused about the names of these classes, `SEC` and `ETF`. We need to come up with some good terminology, but let us discuss that in #531.